### DISCUSSION TITLE
Opening curly brace on a new line in C++ results in the brace being placed in the first column

### DISCUSSION THREAD

**Discussion by ClHg2** [LazyVim/LazyVim#2093] [Q&A]
I have Clangd LazyExtra enabled, and I added the following lines to my `~/.config/nvim/ftplugin/cpp.lua` to set the indentation to 4:

```lua
vim.opt_local.shiftwidth = 4
vim.opt_local.tabstop = 4
```

I'd love to place the opening brace of functions on a new line, but the indentation seems to be incorrect. This is a mininal reproduce(the | symbol means the position of the cursor):

```cpp
class C {
public:
    void update()|
};
```

Then I type `o` to start a new line, it becomes:

```cpp
class C {
public:
    void update()
    |
};
```

Finally, I type `{`, and it becomes this:

```cpp
class C {
public:
    void update()
{|}
};
```

I've read the documents of LazyVim and searched the Internet, but couldn't find the solution. How can I configure neovim to maintain the indentation level when I type symbol `{`?

**Comment by dpetka2001** [comment]
Seems to be some weird `nvim-treesitter` behavior. Disabling treesitter indent with `:TSDisable indent` seems to work like you want. For more help maybe ask in the `nvim-treesitter` Discussions or create an issue if it ends up being a bug?

**Reply by ClHg2** [reply]
Thanks! I'll ask there.
